geth/transactions/factory.go

```diff
@@ -0,0 +1,188 @@
+package transactions
+
+import (
+	"context"
+	"math/big"
+	"sync"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/status-im/status-go/geth/account"
+	"github.com/status-im/status-go/geth/rpc"
+	"github.com/status-im/status-go/sign"
+)
+
+const (
+	// sendTransactionTimeout defines how many seconds to wait before returning result in sentTransaction().
+	sendTransactionTimeout = 300 * time.Second
+	rpcCallTimeout         = time.Minute
+
+	defaultGas = 90000
+)
+
+// Factory validates, signs and propagates transactions to the Ethereum network.
+type Factory struct {
+	signRequests      *sign.PendingRequests
+	ethTxClient       EthTransactor
+	completionTimeout time.Duration
+	rpcCallTimeout    time.Duration
+	networkID         uint64
+
+	addrLock   *AddrLocker
+	localNonce sync.Map
+	log        log.Logger
+}
+
+// NewFactory returns a new Manager.
+func NewFactory(signRequests *sign.PendingRequests) *Factory {
+	return &Factory{
+		signRequests:      signRequests,
+		addrLock:          &AddrLocker{},
+		completionTimeout: sendTransactionTimeout,
+		rpcCallTimeout:    rpcCallTimeout,
+		localNonce:        sync.Map{},
+		log:               log.New("package", "status-go/geth/transactions.Manager"),
```

Comment:
Are we using this approach (injecting logger) throughout the code? I'm not against this approach but it's better to be consistent and use the same approach in all packages (i.e. using global logger vs injecting logger).

Reply:
yeap, we do that since we removed our log package in favour of ethereum log. so we decided to follow their patterns too.